### PR TITLE
UI Scaling for all devices

### PR DESCRIPTION
**Pull request by macsleven**
closes part of #294 

This solves part of scaling UI pages which includes
Splash screen iPad (before)

Splash screen iPad (after):scale component relative to its superview

landing page iPad (before)

landing page iPad (after):scale component relative to its superview

Save seed page(iPad pro 12.9): Button height increases according to screen size and adjusted spacing between seed word.

Save seed page(iPhone 5s)

send page. iPhone 5s(before): send button pushed out of screen and header Text getting chopped off.

send page. iPhone 5s(after) adjusted all components to fit in screen and fixed header text getting chopped off

password /PIN tab bar.ipad(before)

password /PIN tab bar.ipad(after): scale tab bar relative to screen size

security Menu(before)

security Menu(after) fix padding and spacing in PR and make components retain its state.

### CODE REVIEW

**Comment by itswisdomagain** [comment]
Please undo the change to the button height. There is too much spacing/padding around the button text.
> 

The send page now has too much vertical spacing. Vertical spacing between views looks good on master. Send button is also too tall on your PR.
**Master**
Screenshot 2019-04-29 at 3 54 34 PM
**Your PR**
Screenshot 2019-04-29 at 3 54 36 PM

File: decred_wallet/view_controller/SendViewController.swift [hunk 1]
```diff
@@ -13,47 +13,60 @@ import SafariServices
...
-    @IBOutlet weak var exchangeRateError: UIButton!
-    @IBOutlet weak var currencyAmount2: AmountTextfield!
-    
+    weak var delegate: LeftMenuProtocol?
+    @IBOutlet var accountDropdown: DropMenuButton!
+    @IBOutlet weak var toAccountDropDown: DropMenuButton!
+    @IBOutlet weak var toAccountContainer: UIStackView!
+    @IBOutlet weak var toAddressContainer: UIStackView!
+   
+    var fromNotQRScreen = true
```
Comment (itswisdomagain) [inline decred_wallet/view_controller/SendViewController.swift]: I see a lot of variables added here that are not used. Please remove them.

File: decred_wallet/view_controller/WalletSetupViewController.swift [hunk 2]
```diff
@@ -15,6 +15,7 @@ class WalletSetupViewController : UIViewController {
     @IBOutlet weak var restoreWallet: UILabel!
     @IBOutlet weak var createWallet: UILabel!
     @IBOutlet weak var build: UILabel!
+    @IBOutlet weak var walletText: UILabel!
```
Comment (itswisdomagain) [inline decred_wallet/view_controller/WalletSetupViewController.swift]: Delete please

File: decred_wallet/Base.lproj/Main.storyboard [hunk 3]
```diff
@@ -729,10 +735,10 @@
                                 </constraints>
                             </scrollView>
                             <view contentMode="scaleToFill" translatesAutoresizingMaskIntoConstraints="NO" id="wB1-BW-PiR">
-                                <rect key="frame" x="20" y="634" width="335" height="50"/>
+                                <rect key="frame" x="20" y="525.5" width="335" height="64"/>
                                 <subviews>
                                     <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="Failure to keep this seed phrase can result in theft of your funds and inability to recover your wallet." textAlignment="center" lineBreakMode="tailTruncation" numberOfLines="3" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="T26-ed-GAn">
```
Comment (itswisdomagain) [inline decred_wallet/Base.lproj/Main.storyboard]: This has a fixed number of lines. The "write down this seed..." label at the top does not. See how setting a fixed number of lines causes text to cut off on iPhone 4s and causes excess whitespace on iPad 12.9 (screenshots from Interface Builder):

_Also notice the excess spacing between the back button and the page title on the iPad_

Screenshot 2019-05-01 at 4 06 04 AM
Screenshot 2019-05-01 at 4 06 44 AM

File: decred_wallet/view_controller/GeneratedSeedDisplayViewController.swift [hunk 4]
```diff
@@ -11,9 +11,10 @@ import Dcrlibwallet
 
 class GeneratedSeedDisplayViewController: UIViewController {
     
+    @IBOutlet weak var HeaderLabel: UILabel!
+    @IBOutlet weak var subHeaderLabel: UILabel!
```
Comment (itswisdomagain) [inline decred_wallet/view_controller/GeneratedSeedDisplayViewController.swift]: Don't see these 2 new outlets used anywhere in Swift code, please remove.

File: decred_wallet/Base.lproj/Main.storyboard [hunk 5]
```diff
@@ -767,45 +770,87 @@
...
-                            <constraint firstItem="WZx-qh-N33" firstAttribute="trailing" secondItem="qYt-xo-S4a" secondAttribute="trailing" constant="17" id="zyk-qW-OqK"/>
+                            <constraint firstItem="T26-ed-GAn" firstAttribute="width" secondItem="ozH-zT-ShC" secondAttribute="width" multiplier="0.893" id="twg-Y7-hYt"/>
+                            <constraint firstItem="WZx-qh-N33" firstAttribute="bottom" secondItem="2E6-So-256" secondAttribute="bottom" id="x6T-3c-kgn"/>
+                            <constraint firstItem="BQ4-TL-G1r" firstAttribute="top" secondItem="yCc-SR-3I9" secondAttribute="bottom" constant="1" id="zg2-uK-M9V"/>
                         </constraints>
                         <viewLayoutGuide key="safeArea" id="WZx-qh-N33"/>
                     </view>
                     <connections>
+                        <outlet property="HeaderLabel" destination="c6C-Nf-IKq" id="YQd-2H-JBf"/>
+                        <outlet property="buttonCopied" destination="BQ4-TL-G1r" id="umv-YX-TDI"/>
```
Comment (itswisdomagain) [inline decred_wallet/Base.lproj/Main.storyboard]: Don't see the outlet for this, or the view even.

File: decred_wallet/view_controller/Security/SecurityViewController.swift [hunk 6]
```diff
@@ -31,6 +31,9 @@ class SecurityViewController: UIViewController, SeedCheckupProtocol, StartUpPass
         btnPassword.backgroundColor = #colorLiteral(red: 0.9449833035, green: 0.9450840354, blue: 0.9490672946, alpha: 1)
     }
     
+    override func viewDidLoad() {
```
Comment (itswisdomagain) [inline decred_wallet/view_controller/Security/SecurityViewController.swift]: This overriden function does nothing other than calling the same function on the parent class; please remove it.

File: decred_wallet/view_controller/SendViewController.swift [hunk 7]
```diff
@@ -1,11 +1,9 @@
 //
 //  SendViewController.swift
 //  Decred Wallet
-
```
Comment (itswisdomagain) [inline decred_wallet/view_controller/SendViewController.swift]: This is boiler plate code. Please restore the empty line space here.

File: decred_wallet/view_controller/SendViewController.swift [hunk 8]
```diff
@@ -1,11 +1,9 @@
...
 // Copyright (c) 2018-2019 The Decred developers
 // Use of this source code is governed by an ISC
 // license that can be found in the LICENSE file.
-
```
Comment (itswisdomagain) [inline decred_wallet/view_controller/SendViewController.swift]: An empty line space here helps to separate the multi-line comment above from the beginning of code instructions, it's needed. Please restore it.

File: decred_wallet/view_controller/SendViewController.swift [hunk 9]
```diff
@@ -229,40 +227,40 @@ class SendViewController: UIViewController, UITextFieldDelegate,UITextPasteDeleg
...
-         DispatchQueue.main.async {
+        DispatchQueue.main.async {
             if !(self.removedBtn){
-            
-            self.toAddressContainer.removeArrangedSubview(self.pasteBtn)
-            self.pasteBtn.isHidden = true
+                
+                self.toAddressContainer.removeArrangedSubview(self.pasteBtn)
+                self.pasteBtn.isHidden = true
                 //self.removedBtn = true
```
Comment (itswisdomagain) [inline decred_wallet/view_controller/SendViewController.swift]: Commented lines of code and empty white lines can be removed from this if/else blocks.

File: decred_wallet/view_controller/SendViewController.swift [hunk 10]
```diff
@@ -449,10 +447,10 @@ class SendViewController: UIViewController, UITextFieldDelegate,UITextPasteDeleg
             else{
                 confirmSendFundViewController.amount = "\(amountToSend) DCR"
                 confirmSendFundViewController.fee = "\((self.estimateFee.text)!) (\((self.estimateSize.text)!))"
-            
-                }
-            
+
```
Comment (itswisdomagain) [inline decred_wallet/view_controller/SendViewController.swift]: I see a lot of spaces here that have no value. Please delete them.

File: decred_wallet/view_controller/SendViewController.swift [hunk 11]
```diff
@@ -554,17 +552,17 @@ class SendViewController: UIViewController, UITextFieldDelegate,UITextPasteDeleg
...
-            self.showDefaultAccount()
+                self.walletAddress.text = nil
+                self.estimateFee.text = "0.00 DCR"
+                self.estimateSize.text = "0 Bytes"
+                self.BalanceAfter.text = "0.00 DCR"
+                self.tfAmount.text = nil
+                if !(self.toAddressContainer.isHidden){
+                    self.addQrbtn()
+                    self.checkpaste()
+
```
Comment (itswisdomagain) [inline decred_wallet/view_controller/SendViewController.swift]: This space is unnecessary. Don't think it makes sense to have an empty line space before a closing curly brace, any closing curly brace - whether for a function, an if statement, a switch statement; any.

File: decred_wallet/view_controller/SendViewController.swift [hunk 12]
```diff
@@ -624,8 +623,8 @@ class SendViewController: UIViewController, UITextFieldDelegate,UITextPasteDeleg
         return true
     }
     func textFieldShouldClear(_ textField: UITextField) -> Bool {
-         if (textField == self.walletAddress) {
-             DispatchQueue.main.async {
+        if (textField == self.walletAddress) {
+            DispatchQueue.main.async {
                 self.addressErrorText.text = ""
```
Comment (itswisdomagain) [inline decred_wallet/view_controller/SendViewController.swift]: Remove this space please.

File: decred_wallet/view_controller/WalletSetupViewController.swift (hunk 2 above)
Comment (itswisdomagain) [inline decred_wallet/view_controller/WalletSetupViewController.swift]: This outlet is not used, why do we have it?

File: decred_wallet/Base.lproj/Main.storyboard [hunk 13]
```diff
@@ -767,45 +770,87 @@
...
-                                </connections>
-                            </button>
-                            <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="Setting up your wallet" textAlignment="natural" lineBreakMode="tailTruncation" numberOfLines="0" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="c6C-Nf-IKq">
-                                <rect key="frame" x="56" y="73" width="299" height="25.333333333333329"/>
-                                <fontDescription key="fontDescription" type="system" pointSize="21"/>
-                                <color key="textColor" red="0.043137254899999998" green="0.1137254902" blue="0.2470588235" alpha="1" colorSpace="custom" customColorSpace="sRGB"/>
-                                <nil key="highlightedColor"/>
-                            </label>
+                            <view alpha="0.0" contentMode="scaleToFill" translatesAutoresizingMaskIntoConstraints="NO" id="yCc-SR-3I9">
```
[the remark this answers is not in the document]
Comment (macsleven) [reply decred_wallet/Base.lproj/Main.storyboard]: Views are use to adjust spacing between component which may varies across device screen.

File: decred_wallet/view_controller/WalletSetupViewController.swift (hunk 2 above)
Comment (macsleven) [reply decred_wallet/view_controller/WalletSetupViewController.swift]: It was initially added for font scaling in previous commit, but removed now

File: decred_wallet/view_controller/OverviewViewController.swift [hunk 14]
```diff
@@ -611,8 +611,11 @@ DcrlibwalletBlockScanResponseProtocol, DcrlibwalletSpvSyncResponseProtocol,PinEn
            let estimatedDiscoveryTime = totalFetchTime * self.discovery_percentage;
             let totalSyncTime = totalFetchTime + estimatedRescanTime + estimatedDiscoveryTime;
            if totalSyncTime > 0{
-            self.walletInfo.syncRemainingTime = Int64(round(remainingFetchTime + estimatedRescanTime + estimatedDiscoveryTime));
-             self.walletInfo.syncProgress = Int(( Double(elapsedFetchTime) / Double(totalSyncTime) * 100.0))
+            let tmp = remainingFetchTime + estimatedRescanTime + estimatedDiscoveryTime
```
[the remark this answers is not in the document]
Comment (macsleven) [reply decred_wallet/view_controller/OverviewViewController.swift]: moved to a different PR

File: decred_wallet/view_controller/HelpViewController.swift [hunk 15]
```diff
@@ -11,9 +11,11 @@ import SafariServices
 
 class HelpViewController: UIViewController,SFSafariViewControllerDelegate {
     
+    @IBOutlet weak var helpInfo: UILabel!
```
[the remark this answers is not in the document]
Comment (macsleven) [reply decred_wallet/view_controller/HelpViewController.swift]: Added in previous commit for font scaling, Removed now

**Pull request merged**